roomserver/internal/input/input_events.go

```diff
@@ -119,7 +119,7 @@ func (r *Inputer) processRoomEvent(
 		// We haven't calculated a state for this event yet.
 		// Lets calculate one.
 		err = r.calculateAndSetState(ctx, input, *roomInfo, &stateAtEvent, event, isRejected)
-		if err != nil {
+		if err != nil && input.Kind != api.KindOld {
```

Comment:
If err != nil but the kind is kind old we'll fall through which doesn't sound right?

Reply:
Especially as we'll fallthrough without logging `err`

Reply:
Well, this is something I need to think about, because it's entirely possible that we'll fail to work out the state before a `KindOld` event but that doesn't mean we shouldn't then go on to handle redactions etc.